clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -339,11 +339,20 @@ void cleanup() {
         log.trace("Closing the consumer network thread");
         Timer timer = time.timer(closeTimeout);
         try {
-            runAtClose(requestManagers.entries(), networkClientDelegate, time.milliseconds());
+            // If an error was thrown from initializeResources(), it's possible that the list of request managers
+            // is null, so check before using. If the request manager list is null, there wasn't any real work
+            // performed, so not being able to close the request managers isn't so bad.
+            if (requestManagers != null)
+                runAtClose(requestManagers.entries(), networkClientDelegate, time.milliseconds());
```

Comment:
Should we add null check for `networkClientDelegate` too?

Reply:
Done. I’d convinced myself that because the `RequestManagers` instance is initialized _after_ the `NetworkClientDelegate` instance in `initializeResources()` that it wasn't strictly necessary. But it's cleaner and more defensive this way 👍